crates/nu-protocol/src/value/primitive.rs

```diff
@@ -89,29 +87,9 @@ impl Primitive {
     // FIXME: This is a bad name, but no other way to differentiate with our own Duration.
     pub fn into_chrono_duration(self, span: Span) -> Result<chrono::Duration, ShellError> {
         match self {
-            Primitive::Duration(duration) => {
-                let (secs, nanos) = duration.div_rem(&BigInt::from(NANOS_PER_SEC));
-                let secs = match secs.to_i64() {
-                    Some(secs) => secs,
-                    None => {
-                        return Err(ShellError::labeled_error(
-                            "Internal duration conversion overflow.",
-                            "duration overflow",
-                            span,
-                        ))
-                    }
-                };
-                // This should never fail since nanos < 10^9.
-                let nanos = match nanos.to_i64() {
-                    Some(nanos) => nanos,
-                    None => return Err(ShellError::unexpected("Unexpected i64 overflow")),
-                };
-                let nanos = chrono::Duration::nanoseconds(nanos);
-                // This should also never fail since we are adding less than NANOS_PER_SEC.
-                chrono::Duration::seconds(secs)
-                    .checked_add(&nanos)
-                    .ok_or_else(|| ShellError::unexpected("Unexpected duration overflow"))
-            }
+            Primitive::Duration(duration) => Ok(chrono::Duration::nanoseconds(
+                duration.to_i64().expect("Unexpected integer overflow"),
```

Comment:
This looks like it shortens nanos to i64. Currently, we store it as a BigInt so that we can handle values larger than i64.

Are you able to simplify but not lose this additional information?

Reply:
I see, so this update would crash in the event that a BigInt is larger than `2 ^ 64`, which is somewhere around the year `2554` in nanoseconds. I'm guessing we should keep as-is to be more robust?

Reply:
Set it back, I will think of a way of simplfying, but may leave it for another commit/PR. 

I believe the `from` inverse function should be safe since a `chrono::Duration` shouldn't be more than i64